components/widgets/inputs/dropdown_input_hybrid.tsx

```diff
@@ -0,0 +1,253 @@
+    inputType?: string;
+    defaultValue: T;
+};
+
+const baseStyles = {
+    input: (provided: CSSProperties) => ({
+        ...provided,
+        color: 'var(--center-channel-color)',
+    }),
+    control: (provided: CSSProperties) => ({
+        ...provided,
+        border: 'none',
+        boxShadow: 'none',
+        padding: '0 2px',
+        cursor: 'pointer',
+    }),
+    indicatorSeparator: (provided: CSSProperties) => ({
+        ...provided,
+        display: 'none',
+    }),
+    menuPortal: (provided: CSSProperties) => ({
+        ...provided,
+        zIndex: 99999999,
+    }),
+};
+
+
+const IndicatorsContainer = (props: IndicatorContainerProps<ValueType>) => (
+    <div className='DropdownInput__indicatorsContainer'>
+        <components.IndicatorsContainer {...props}>
+            <i className='icon icon-chevron-down'/>
+        </components.IndicatorsContainer>
+    </div>
+);
+
+const Control = (props: ControlProps<ValueType>) => (
+    <div className='DropdownInput__controlContainer'>
+        <components.Control {...props}/>
+    </div>
+);
+
+const Option = (props: OptionProps<ValueType>) => (
+    <div
+        className={classNames('DropdownInput__option', {
+            selected: props.isSelected,
+            focused: props.isFocused,
+        })}
+    >
+        <components.Option {...props}/>
+    </div>
+);
+
+const DropdownInputHybrid = <T extends ValueType>(props: Props<T>) => {
+    const {
+        value,
+        placeholder,
+        className,
+        name,
+        legend, 
+        onDropdownChange,
+        onInputChange,
+        error,
+        exceptionToInput,
+        width,
+        inputValue,
+        inputType,
+        defaultValue,
+        ...otherProps
+    } = props;
+
+    const containerRef = useRef<HTMLInputElement>(null);
+    const inputRef = useRef<HTMLInputElement>(null);
+    const [inputFocused, setInputFocused] = useState(false);
+    const [focused, setFocused] = useState(false);
+    const [showInput, setShowInput] = useState(!!inputValue);
+
+    useEffect(() => {
+        if (showInput && !inputValue && inputRef.current) {
+            inputRef.current.focus();
+        }
+    }, [showInput]);
+
+    useEffect(() => {
+        if (!inputFocused) {
+            showTextInput(inputValue, false);
+        }
+    }, [inputValue]);
+
+    useEffect(() => {
+        if (!inputValue && !focused && !inputFocused) {
+            onDropdownChange(props.defaultValue);
+            showTextInput('');
+        }
+    }, [focused, inputFocused]);
+
+    const getMenuStyles = () => {
+        if (showInput) {
+            return {
+                menu: (provided: CSSProperties) => ({
+                    ...provided,
+                    width: containerRef.current ? `${containerRef.current.offsetWidth}px` : '0px',
+                    left: inputRef.current ? `-${inputRef.current.offsetWidth}px` : '0px',
+                }),
+            };
+        }
+        return {};
+    }
```

Comment:
I personally don't like conditional returns (I would rather do it before the `getMenuStyles` invocation to prevent invocation altogether when `showInput` is not set), but in this case it would probably simplify it by adding a ternary instead of a complete if-block

```suggestion
    const getMenuStyles = () =>
        showInput
            ? {
                  menu: (provided: CSSProperties) => ({
                      ...provided,
                      width: containerRef.current ? `${containerRef.current.offsetWidth}px` : '0px',
                      left: inputRef.current ? `-${inputRef.current.offsetWidth}px` : '0px',
                  }),
              }
            : {};
```